fn main() {
    1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1 = 1;
// { dg-error ".E0070." "" { target *-*-* } .-1 }
// { dg-error ".E0070." "" { target *-*-* } .-2 }
// { dg-error ".E0070." "" { target *-*-* } .-3 }
// { dg-error ".E0070." "" { target *-*-* } .-4 }
// { dg-error ".E0070." "" { target *-*-* } .-5 }
// { dg-error ".E0070." "" { target *-*-* } .-6 }
// { dg-error ".E0070." "" { target *-*-* } .-7 }
// { dg-error ".E0070." "" { target *-*-* } .-8 }
// { dg-error ".E0070." "" { target *-*-* } .-9 }
// { dg-error ".E0070." "" { target *-*-* } .-10 }
// { dg-error ".E0070." "" { target *-*-* } .-11 }
// { dg-error ".E0070." "" { target *-*-* } .-12 }
// { dg-error ".E0070." "" { target *-*-* } .-13 }
// { dg-error ".E0070." "" { target *-*-* } .-14 }
// { dg-error ".E0070." "" { target *-*-* } .-15 }
// { dg-error ".E0070." "" { target *-*-* } .-16 }
// { dg-error ".E0070." "" { target *-*-* } .-17 }
// { dg-error ".E0070." "" { target *-*-* } .-18 }
// { dg-error ".E0070." "" { target *-*-* } .-19 }
// { dg-error ".E0070." "" { target *-*-* } .-20 }
// { dg-error ".E0070." "" { target *-*-* } .-21 }
// { dg-error ".E0070." "" { target *-*-* } .-22 }
// { dg-error ".E0070." "" { target *-*-* } .-23 }
}

