macro_rules! {}
// { dg-error "" "" { target *-*-* } .-1 }
// { dg-note "" "" { target *-*-* } .-2 }
// { dg-error "" "" { target *-*-* } .-3 }

macro_rules!{}
// { dg-error "" "" { target *-*-* } .-1 }
// { dg-note "" "" { target *-*-* } .-2 }
// { dg-error "" "" { target *-*-* } .-3 }

fn main() {}

