fn foo(x: impl async Fn()) -> impl async Fn() { x }
// { dg-error ".E0658." "" { target *-*-* } .-1 }
// { dg-error ".E0658." "" { target *-*-* } .-2 }
// { dg-error ".E0658." "" { target *-*-* } .-3 }
// { dg-error ".E0658." "" { target *-*-* } .-4 }
// { dg-error ".E0658." "" { target *-*-* } .-5 }
// { dg-error ".E0658." "" { target *-*-* } .-6 }

fn main() {}

