#![allow(unused)]

fn test_122112() {
    // Make sure we don't ICE if parsing in recovery fails
    let _: std::env::temp_dir().join(&self, push: Box<usize>);  // { dg-error "" "" { target *-*-* } }
}

fn main() {
    let _: std::env::temp_dir().join("foo"); // { dg-error "" "" { target *-*-* } }
}

