fn main() -> Result<(), ()> {
    a(b(c(d(e(
// { dg-error ".E0425." "" { target *-*-* } .-1 }
// { dg-error ".E0425." "" { target *-*-* } .-2 }
// { dg-error ".E0425." "" { target *-*-* } .-3 }
// { dg-error ".E0425." "" { target *-*-* } .-4 }
// { dg-error ".E0425." "" { target *-*-* } .-5 }
        z????????????????????????????????????????????????????????????????????????????????????????
        ?????????????????????????????????????????????????????????????????????????????????????????
        ??????????????????????????????????????????????????????????????????
// { dg-error ".E0425." "" { target *-*-* } .-3 }
    )))))
}

