fn main() {
    let str::<{fn str() { let str::T>>::as_bytes; }}, T>::as_bytes;
// { dg-error ".E0282." "" { target *-*-* } .-1 }
// { dg-error ".E0282." "" { target *-*-* } .-2 }
// { dg-error ".E0282." "" { target *-*-* } .-3 }
// { dg-error ".E0282." "" { target *-*-* } .-4 }
// { dg-error ".E0282." "" { target *-*-* } .-5 }
}

