#[derive(unsafe(Debug))]
// { dg-error "" "" { target *-*-* } .-1 }
// { dg-error "" "" { target *-*-* } .-2 }
// { dg-error "" "" { target *-*-* } .-3 }
// { dg-error "" "" { target *-*-* } .-4 }
// { dg-error "" "" { target *-*-* } .-5 }
// { dg-error "" "" { target *-*-* } .-6 }
struct Foo;

#[unsafe(derive(Debug))] // { dg-error "" "" { target *-*-* } }
struct Bar;

fn main() {}

