#![feature(type_ascription)]

fn e() {
    type_ascribe!(p, a<p:p<e=6>>);
// { dg-error ".E0658." "" { target *-*-* } .-1 }
// { dg-error ".E0658." "" { target *-*-* } .-2 }
// { dg-error ".E0658." "" { target *-*-* } .-3 }
// { dg-error ".E0658." "" { target *-*-* } .-4 }
}

fn main() {}

